flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/rank/RetractableTopNFunctionTest.java

```diff
@@ -209,12 +301,12 @@ public void testDisableGenerateRetraction() throws Exception {
 		List<Object> expectedOutput = new ArrayList<>();
 		expectedOutput.add(record("book", 1L, 12));
 		expectedOutput.add(record("book", 2L, 19));
-		expectedOutput.add(record("book", 1L, 12));
 		expectedOutput.add(record("book", 4L, 11));
+		expectedOutput.add(deleteRecord("book", 2L, 19));
```

Comment:
We should emit delete/retract record first, then emit accumulate record. Otherwise, there might be some problems, e.g. user may see instantly incorrect results (top3 but see top4 at some point of time).

Reply:
make sense.